pandas/tslib.pyx

```diff
@@ -1220,7 +1220,10 @@ cdef class _Timestamp(datetime):
             return Timestamp((self.freq * other).apply(self), freq=self.freq)
 
         elif isinstance(other, timedelta) or hasattr(other, 'delta'):
-            nanos = _delta_to_nanoseconds(other)
+            try:
```

Comment:
cc @shoyer 
since IIRC you have some expertise doing this.

I could not get to raise the exception at all, rather it always wanted to simply call ``__radd__``. The only way is to return ``NotImplemented``.

any better way to do this?